### PR TITLE
Convert SpatialMaterial references to StandardMaterial3D.

### PR DESCRIPTION
**Pull request by madmiraal**
Fixes broken links to `class_spatialmaterial`.

`SpatialMaterial` has been replaced with `StandardMaterial3D`. For consistency, I've renamed all references to SpatialMaterial to StandardMaterial3D even if the underlying documentation will need to be updated further.

I've also renamed `tutorials/3d/spatial_material.rst` to `tutorials/3d/standard_material_3d.rst`. Note, https://github.com/godotengine/godot/blob/master/doc/classes/BaseMaterial3D.xml currently references the old file; so I'll submit a separate PR to fix that.

**Edit**: Updated to convert references from `SpatialMaterial` to `StandardMaterial3D`.

### CODE REVIEW

**Comment by clayjohn** [comment]
This may take a little rethinking as users cannot use ``BaseMaterial3D`` directly, it is a parent class for ``StandardMaterial3D`` and ``ORMMaterial3D``. 

Further this PR conflicts with https://github.com/godotengine/godot-docs/pull/3401

**Comment by madmiraal** [comment]
I've updated the references to use StandardMaterial3D instead.

File: tutorials/misc/gles2_gles3_differences.rst [hunk 1]
```diff
@@ -50,20 +50,20 @@ GLES2 is not capable of using High Dynamic Range (HDR) rendering features. If HD
...
 - Subsurface scattering
 - Anisotropy
 - Clearcoat
 - Depth mapping
 
-When using SpatialMaterials they will not even appear in the editor.
+When using StandardMaterial3Ds they will not even appear in the editor.
 
-In custom :ref:`ShaderMaterials <class_ShaderMaterial>`, you can set values for these features but they
+In custom :ref:`class_StandardMaterial3D <class_StandardMaterial3D>`, you can set values for these features but they
```
Comment (clayjohn) [inline tutorials/misc/gles2_gles3_differences.rst:66]: There are a few false positives like this one. ShaderMaterials still exist and are very different from StandardMaterial3Ds. So all references to ShaderMaterials should be reverted back.

Comment (madmiraal) [reply tutorials/misc/gles2_gles3_differences.rst]: I've fixed it and double-checked that I haven't done it elsewhere too.

**Comment by madmiraal** [comment]
Rebased following #3482.

**Comment by mhilbrunner** [comment]
Looks good to me, thanks for the rebase :)

Thank you!

**Pull request merged**